src/transformers/models/sam/image_processing_sam.py

```diff
@@ -1132,9 +1134,10 @@ def _generate_crop_boxes(
     )
     crop_boxes = np.array(crop_boxes)
     crop_boxes = crop_boxes.astype(np.float32)
-    points_per_crop = np.array([point_grid_per_crop])
-    points_per_crop = np.transpose(points_per_crop, axes=(0, 2, 1, 3))
 
+    # Reshape points to match the expected format for batched processing
+    points_per_crop = np.array(point_grid_per_crop)
+    points_per_crop = np.expand_dims(points_per_crop, axis=1)  # Add batch dimension
```

Comment:
Can `points_per_crop` be a batch already? Should we check ndim before adding another one?